src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,103 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useCallback } from 'react';
+
+export enum FilterTodoStatus {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+export interface IFilter {
+  status: FilterTodoStatus;
+  searchTitle: string;
+}
+
+export type OnFilterChange = (filter: IFilter) => void;
+interface Props {
+  filter: IFilter;
+  onChange: OnFilterChange;
+}
+
+export const TodoFilter: React.FC<Props> = ({ onChange, filter }) => {
+  const handleStatusChanged = useCallback<
+  React.ChangeEventHandler<HTMLSelectElement>
+  >(
+    (event) => {
+      if (onChange) {
+        const status = event.target.value as FilterTodoStatus;
```

Comment:
try to avoid `as` in typeScript, bc you say him - i know that there will be FilterTodoStatus, but if you put to the wrong input - you will have crash, but TS don't know about that